oidc/src/main/java/io/quarkiverse/renarde/oidc/impl/RernardeRevokeController.java

```diff
@@ -0,0 +1,82 @@
+package io.quarkiverse.renarde.oidc.impl;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Base64;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.Response;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import io.quarkiverse.renarde.Controller;
+import io.quarkiverse.renarde.security.RenardeSecurity;
+import io.quarkus.oidc.AccessTokenCredential;
+import io.quarkus.security.Authenticated;
+import io.smallrye.jwt.algorithm.SignatureAlgorithm;
+import io.smallrye.jwt.build.Jwt;
+
+@Path("_renarde/security")
+public class RernardeRevokeController extends Controller {
+
+    @Inject
+    AccessTokenCredential accessToken;
+
+    @RestClient
+    RenardeAppleClient renardeAppleClient;
+
+    @ConfigProperty(name = "quarkus.oidc.apple.client-id")
+    String appleClientId;
+
+    @ConfigProperty(name = "quarkus.oidc.apple.credentials.jwt.issuer")
+    String appleOidcIssuer;
+
+    @ConfigProperty(name = "quarkus.oidc.apple.credentials.jwt.token-key-id")
+    String appleOidcKeyId;
+
+    @ConfigProperty(name = "quarkus.oidc.apple.credentials.jwt.key-file", defaultValue = "AuthKey_XXX.p8")
+    String appleKeyFile;
+
+    @Inject
+    public RenardeSecurity security;
+
+    @Path("apple-revoke")
+    @Authenticated
+    public Response revokeApple() {
+        String clientSecret = Jwt.audience("https://appleid.apple.com")
+                .subject(appleClientId)
+                .issuer(appleOidcIssuer)
+                .issuedAt(Instant.now().getEpochSecond())
+                .expiresIn(Duration.ofHours(1))
+                .jws()
+                .keyId(appleOidcKeyId)
+                .algorithm(SignatureAlgorithm.ES256)
+                .sign(getPrivateKey(appleKeyFile));
+        renardeAppleClient.revokeAppleUser(appleClientId, clientSecret, accessToken.getToken(), "access_token");
+        return security.makeLogoutResponse();
+    }
+
+    private PrivateKey getPrivateKey(String filename) {
+        try {
+            String content = Files.readString(Paths.get("target/classes/" + filename));
```

Comment:
I'm pretty sure this is where we should fetch it from. In fact, for native compiled or packaged production applications, this folder will not exist. This will only work in DEV mode, I think.

Reply:
I've tried using `quarkus-run.jar`and it was ok but you must be right. Unfortunatelly I didn't find a better solution. I guess quarkus-oidc lib got something  to help on that as it already read this file. WDYT ?